test/honeycomb_cth.erl

```diff
@@ -0,0 +1,101 @@
+-module(honeycomb_cth).
+
+-export([id/1]).
+-export([init/2]).
+
+-export([pre_init_per_testcase/4]).
+-export([post_end_per_testcase/5]).
+
+-record(state, {directory, github_workflow, github_run_id,
+                github_repository, github_sha, github_ref,
+                base_rmq_ref, erlang_version, elixir_version,
+                otp_release, cpu_topology_json, schedulers,
+                system_architecture, system_memory_data_json,
+                start_times = #{}}).
+
+id(Opts) ->
+    proplists:get_value(directory, Opts, "/tmp/honeycomb").
+
+init(Id, _Opts) ->
+    application:ensure_all_started(os_mon),
+    {ok, #state{directory = Id,
+                github_workflow = os:getenv("GITHUB_WORKFLOW", "unknown"),
+                github_run_id = os:getenv("GITHUB_RUN_ID", "unknown"),
+                github_repository = os:getenv("GITHUB_REPOSITORY", "unknown"),
+                github_sha = os:getenv("GITHUB_SHA", "unknown"),
+                github_ref = os:getenv("GITHUB_REF", "unknown"),
+                base_rmq_ref = os:getenv("BASE_RMQ_REF", "unknown"),
+                erlang_version = os:getenv("ERLANG_VERSION", "unknown"),
+                elixir_version = os:getenv("ELIXIR_VERSION", "unknown"),
+                otp_release = erlang:system_info(otp_release),
+                cpu_topology_json = cpu_topology_json(erlang:system_info(cpu_topology)),
+                schedulers = erlang:system_info(schedulers),
+                system_architecture = erlang:system_info(system_architecture),
+                system_memory_data_json = json_string_memory(memsup:get_system_memory_data())}}.
+
+pre_init_per_testcase(_Suite, TC, Config, #state{start_times = StartTimes} = State) ->
+    {Config, State#state{start_times = StartTimes#{TC => erlang:timestamp()}}}.
```

Comment:
This start times map is created only on init. It should probably be created per-suite to avoid testcase name clashes

Reply:
I have corrected this